examples/pytorch/summarization/run_summarization_no_trainer.py

```diff
@@ -671,7 +668,7 @@ def postprocess_text(preds, labels):
             args.val_max_target_length = args.max_target_length
 
         gen_kwargs = {
-            "max_length": args.val_max_target_length if args is not None else config.max_length,
+            "max_length": args.max_length if args.max_length is not None else args.val_max_target_length,
```

Comment:
It should be the `max_target_length` here if `val_max_target_length` is `None`, `max_length` is for the inputs only.

Reply:
So you can just use `max_target_length` here or the config max_length if it was set to None.

Reply:
`max_target_length` is also used in `preprocess_function` to truncate the outputs. 

Here I think it might be better to provide another generation max length parameter, which is only used for evaluation/prediction, as done in examples/pytorch/summarization/run_summarization.py. In case the users don't want to truncate the summaries in dataset, but want to limit the generation max length in evaluation to accelerate the whole training process?

And of course if it is `None`, it will be set to the `val_max_target_length` (or `max_target_length`) or the `max_length` in model's config

Reply:
I don't think so. The parameter comes from a copy-paste and wasn't used, I wouldn't find a use for it. If the user wants some none-standard behavior, they should adapt the example.